src/auth/auth.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import * as bcrypt from 'bcrypt';
+
+import { UserService } from '@src/user/user.service';
+import { CreateUserDto } from '@src/user/dto/create-user.dto';
+import { LoginUserDto } from './dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UserService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async register(createUserDto: CreateUserDto) {
+    const user = await this.userService.create(createUserDto);
+
+    return { ...user, token: this.generateJwtToken(user.id) };
+  }
+
+  async login(loginDto: LoginUserDto) {
+    const { email, password } = loginDto;
+    const user = await this.userService.findOne(email);
+
+    if (!user) {
+      throw new UnauthorizedException('Invalid credentials');
+    }
+
+    if (!this.isValidPassword(password, user.password))
+      throw new UnauthorizedException('Invalid credentials');
+
+    delete user.password;
```

Comment:
It would be better to set this at the user entity. You can use the package class-transformer.
In the user entity, the password attribute should have the Exclude decorator.

```
@Column({ name: 'password' })
@Exclude({ toPlainOnly: true })
password: string;
```

Reply:
the problem with that solution is that I don't get the password back and I need for the validation